src/target/riscv/riscv.c

```diff
@@ -5031,6 +5005,51 @@ COMMAND_HANDLER(riscv_set_enable_trigger_feature)
 	return ERROR_OK;
 }
 
+static COMMAND_HELPER(report_reserved_triggers, struct target *target)
+{
+	RISCV_INFO(r);
+	if (riscv_enumerate_triggers(target) != ERROR_OK)
+		return ERROR_FAIL;
+	const char *separator = "";
+	for (riscv_reg_t t = 0; t < r->trigger_count; ++t)
+		if (r->reserved_triggers[t]) {
+			command_print_sameline(CMD, "%s%" PRIu64, separator, t);
+			separator = " ";
+		}
+	command_print_sameline(CMD, "\n");
+	return ERROR_OK;
+}
+
+COMMAND_HANDLER(handle_reserve_trigger)
+{
+	struct target *target = get_current_target(CMD_CTX);
+	if (CMD_ARGC == 0)
+		return CALL_COMMAND_HANDLER(report_reserved_triggers, target);
+
+	if (CMD_ARGC != 2)
+		return ERROR_COMMAND_SYNTAX_ERROR;
+
+	riscv_reg_t t;
+	COMMAND_PARSE_NUMBER(u64, CMD_ARGV[0], t);
+
+	if (riscv_enumerate_triggers(target) != ERROR_OK)
+		return ERROR_FAIL;
+	RISCV_INFO(r);
+	if (t >= r->trigger_count) {
+		command_print(CMD, "Error: trigger %" PRIu64
+				" does not exist, there are only %u triggers on the target.",
+				t, r->trigger_count);
+		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
+	}
```

Comment:
Optional detail:

I tried to make the trigger index more clear to the user - the 0..N-1 range. 

```suggestion
	if (r->trigger_count == 0) {
		command_print(CMD, "Error: There are no triggers in the target.");
		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
	}
	if (t >= r->trigger_count) {
		command_print(CMD, "Error: trigger with index %" PRIu64
				" does not exist. There are only %u triggers on the target (with indexes 0 .. %u).",
				t, r->trigger_count, r->trigger_count - 1);
		return ERROR_TARGET_RESOURCE_NOT_AVAILABLE;
	}
```

Reply:
Addressed.